### DISCUSSION TITLE
NDI output option

### DISCUSSION THREAD

**Discussion by michaelatwyes** [cpvalente/ontime#1286] [Ideas]
Would it be possible to include NDI as an output option for the different viewers?

**Comment by cpvalente** [comment]
> Would it be possible to include NDI as an output option for the different viewers? 

Hi!

This has come up a few times. 

We have generally agreed that we did not want to enlarge the scope of Ontime with an NDI streaming feature.

From our talk with users [vingester](https://vingester.app/) has been suggested as a lightweight app that produces niceties like CSS override. 

I appreciate that users want to reduce the amount of apps they use and I hope this does not discourage from future suggestions

**Comment by ImpossibleWorks** [comment]
Would love to add a +1 for this feature. Not having a dedicated laptop to display a timer screen and convert to NDI would be super helpful ;-) considering vingester looks abandoned.

**Reply by alex-Arc** [reply]
Have you tried this
https://www.tractusevents.com/tools/html-to-ndi/
Haven't tested myself yet

**Comment by ImpossibleWorks** [comment]
Thanks for sharing that. I haven't tried it since I'm Mac-based. Looks like a great tool, yet it would require another computer onsite to perform that task. ;-)

**Reply by Punkerschaf** [reply]
You can use CasparCG to achieve NDI outputs of Ontime. I've done this with both NDI and SDI.
It's not very user friendly while it's configuration is done via XML and for this use case the UI client is not handy.

But you can configure multiple outputs each with a so called "NDI consumer". In addition you hardcode browser sources (called "producers") into the server config, so it's directly loaded when starting the server.
Theoretically you can also use CasparCG client to change or restart single sources.

EDIT: Oh sorry, I overred that you are Mac-based. So CasparCG doesn't work while there are currently only Win and Linux builds.

**Comment by alex-Arc** [comment]
> yet it would require another computer onsite to perform that task

Are you using ontime cloud? In that case you would not get any use from it
If you are running ontime locally you can just run it on the same machine no?

and as you maybe can see from the [tractus.HtmlToNdi](https://github.com/tractusevents/tractus.HtmlToNdi/) and [vingester ](https://github.com/rse/vingester)projects it is not a simple thing we can just add unfortunately
